src/drivers/intel/dmic/dmic_nhlt.c

```diff
@@ -235,7 +248,10 @@ int dmic_set_config_nhlt(struct dai *dai, void *spec_config)
 		return -EINVAL;
 	}
 
-	for (n = 0; n < num_fifos; n++) {
+	for (n = 0; n < DMIC_HW_FIFOS_MAX; n++) {
```

Comment:
actually do we need this loop - do we need to verify both FIFOs? Shouldn't we only verify the one for `dai->index`?

Reply:
I think it's good to error early about a corrupt blob when the data is available. But it's not a big deal, what do you prefer?

Reply:
This configuration data is passed as a part of the `SOF_IPC_DAI_CONFIG` IPC processing for a specific DMIC DAI, but does the configuration data contain both DMICs?

Reply:
Yes, the blob contains configuration for both FIFOs/DAIs.